core/src/main/java/jenkins/mvn/SettingsProvider.java

```diff
@@ -21,8 +24,28 @@
 public abstract class SettingsProvider extends AbstractDescribableImpl<SettingsProvider> implements ExtensionPoint {
 
     /**
+     * <p>
+     *     Configure maven launcher argument list with adequate settings path.
+     * </p>
+     * <p>Implementations should
+     * <ul>Be aware that this method might get called multiple times during a build.</ul>
+     * <ul>Implement this method. This class provides a default implementation throwing an {@link UnsupportedOperationException}
+     * so that implementations have time to adapt.</ul>
+     * </p>
+     *
+     * @param run       the build / run to provide the settings for
+     * @param workspace the workspace in which the build / run takes place
+     * @param listener the listener of this given build / run
+     * @return the filepath to the provided file. <code>null</code> if no settings will be provided.
+     */
+    @CheckForNull
+    public FilePath supplySettings(@Nonnull Run<?, ?> run,  @Nonnull FilePath workspace, @Nonnull TaskListener listener) {
+        throw new UnsupportedOperationException("Maven SettingsProvider " + getClass() + " does not yet support injecting Maven settings in jobs of type " + run.getClass());
```

Comment:
🐛 The implementation should check `instanceof AbstractBuild` and invoke old method by default. Otherwise it should return null and maybe log warning